pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -77,40 +77,35 @@ var (
 		Effect: v1.TaintEffectNoExecute,
 	}
 
-	nodeConditionToTaintKeyStatusMap = map[v1.NodeConditionType]struct {
-		taintKey string
-		// noScheduleStatus is the condition under which the node should be tainted as not schedulable for this
-		// NodeConditionType
-		noScheduleStatus v1.ConditionStatus
-	}{
+	// map {NodeConditionType: {ConditionStatus: TaintKey}}
+	// represents which NodeConditionType under which ConditionStatus should be
+	// tainted with which TaintKey
+	// for certain NodeConditionType, there are multiple {ConditionStatus,TaintKey} pairs
+	nodeConditionToTaintKeyStatusMap = map[v1.NodeConditionType]map[v1.ConditionStatus]string{
 		v1.NodeReady: {
-			taintKey:         algorithm.TaintNodeNotReady,
-			noScheduleStatus: v1.ConditionFalse,
+			v1.ConditionFalse:   algorithm.TaintNodeNotReady,
+			v1.ConditionUnknown: algorithm.TaintNodeUnreachable,
```

Comment:
Does it mean NodeCondition with type = `NodeReady` and status = `ConditionUnknown` should be considered as `NodeUnreachable`?

It is a little ambiguous. Perhaps a new type `NodeReachable` is more reasonable.
https://github.com/kubernetes/kubernetes/blob/5a16163c87fe2a90916a51b52771a668bcaf2a0d/staging/src/k8s.io/api/core/v1/types.go#L4019-L4023

Reply:
> Does it mean NodeCondition with type = NodeReady and status = ConditionUnknown should be considered as NodeUnreachable?

yeap.



> It is a little ambiguous. Perhaps a new type NodeReachable is more reasonable.

Agree, that's the ideal case. I would also like to see this kind of status :)

But that should be considered by sig/node, maybe as a new feature.